src/libraries/System.Private.CoreLib/src/System/Buffers/SharedArrayPool.cs

```diff
@@ -107,7 +107,9 @@ public override T[] Rent(int minimumLength)
                 ArgumentOutOfRangeException.ThrowIfNegative(minimumLength);
             }
 
-            buffer = new T[minimumLength];
+            buffer = typeof(T).IsPrimitive && typeof(T) != typeof(bool) ?
+                GC.AllocateUninitializedArray<T>(minimumLength) :
+                new T[minimumLength];
```

Comment:
```suggestion
            // For large arrays, we prefer to avoid the zero-initialization costs. However, as the resulting
            // arrays could end up containing arbitrary bit patterns, we only allow this for types for which
            // every possible bit pattern is valid.
            buffer = typeof(T).IsPrimitive && typeof(T) != typeof(bool) ?
                GC.AllocateUninitializedArray<T>(minimumLength) :
                new T[minimumLength];

```